contracts/cw4-group/helpers.ts

```diff
@@ -0,0 +1,267 @@
+    } catch (err) {
+      // generate if no file exists
+      const wallet = await DirectSecp256k1HdWallet.generate(12, {hdPaths: [options.hdPath], prefix: options.bech32prefix});
+      const encrypted = await wallet.serialize(password);
+      fs.writeFileSync(filename, encrypted, 'utf8');
+      return wallet;
+    }
+    // otherwise, decrypt the file (we cannot put deserialize inside try or it will over-write on a bad password)
+    const wallet = await DirectSecp256k1HdWallet.deserialize(encrypted, password);
+    return wallet;
+  };
+
+  const connect = async (
+    wallet: DirectSecp256k1HdWallet,
+    options: Options
+  ): Promise<SigningCosmWasmClient> => {
+    const clientOptions = {
+      prefix: options.bech32prefix
+    }
+    return await SigningCosmWasmClient.connectWithSigner(options.httpUrl, wallet, clientOptions)
+  };
+
+  const hitFaucet = async (
+    faucetUrl: string,
+    address: string,
+    denom: string
+  ): Promise<void> => {
+    await axios.post(faucetUrl, {denom, address});
+  }
+
+  const setup = async (password: string, filename?: string): Promise<[string, SigningCosmWasmClient]> => {
+    const keyfile = filename || options.defaultKeyFile;
+    const wallet = await loadOrCreateWallet(pebblenetOptions, keyfile, password);
+    const client = await connect(wallet, pebblenetOptions);
+
+    const [account] = await wallet.getAccounts();
+    // ensure we have some tokens
+    if (options.faucetUrl) {
+      const tokens = await client.getBalance(account.address, options.feeToken)
+      if (tokens.amount === '0') {
+        console.log(`Getting ${options.feeToken} from faucet`);
+        await hitFaucet(options.faucetUrl, account.address, options.feeToken);
+      }
+    }
+
+    return [account.address, client];
+  }
+
+  const recoverMnemonic = async (password: string, filename?: string): Promise<string> => {
+    const keyfile = filename || options.defaultKeyFile;
+    const wallet = await loadOrCreateWallet(pebblenetOptions, keyfile, password);
+    return wallet.mnemonic;
+  }
+
+  return {setup, recoverMnemonic};
+}
+
+interface AdminResponse {
+  readonly admin?: string
+}
+
+interface MemberResponse {
+  readonly weight?: number;
+}
+
+interface MemberListResponse {
+  readonly members: number;
+}
+
+interface TotalWeightResponse {
+  readonly weight: number;
+}
+
+interface HooksResponse {
+  readonly hooks: readonly string[];
+}
+
+interface CW4GroupInstance {
+  readonly contractAddress: string
+
+  // queries
+  admin: () => Promise<AdminResponse>
+  totalWeight: () => Promise<TotalWeightResponse>
+  member: (addr: string, atHeight?: number) => Promise<MemberResponse>
+  listMembers: (startAfter?: string, limit?: number) => Promise<MemberListResponse>
+  hooks: () => Promise<HooksResponse>
+
+  // actions
+  updateAdmin: (txSigner: string, admin?: string) => Promise<string>
+  updateMembers: (txSigner: string, remove: string[], add: string[] ) => Promise<string>
```

Comment:
See:

```rust
    UpdateMembers {
        remove: Vec<String>,
        add: Vec<Member>,
    },
```

The add argument is a `[]Member` like above in the query... it must have both the address and the weight